pkg/virt-controller/services/template.go

```diff
@@ -374,10 +398,19 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 	containers = append(containers, container)
 
 	for i, requestedHookSidecar := range requestedHookSidecarList {
+		resources := k8sv1.ResourceRequirements{}
+		// add default cpu and memory limits to enable cpu pinning if requested
+		// TODO(vladikr): make the hookSidecar express resources
+		if vmi.IsCPUDedicated() {
+			resources.Limits = make(k8sv1.ResourceList)
+			resources.Limits[k8sv1.ResourceCPU] = resource.MustParse("200m")
+			resources.Limits[k8sv1.ResourceMemory] = resource.MustParse("64M")
+		}
 		containers = append(containers, k8sv1.Container{
 			Name:            fmt.Sprintf("hook-sidecar-%d", i),
 			Image:           requestedHookSidecar.Image,
 			ImagePullPolicy: requestedHookSidecar.ImagePullPolicy,
+			Resources:       resources,
```

Comment:
@vladikr This super minor, but maybe worth thinking about.

Since we're defaulting all the sidecar and registry disk container resource limits to 200m/64M, would it make more sense to just iterate over the Pod's container list at the end of this function and force the 200m/64M limit on any container that isn't the compute container?

That would also enforce that limits are required in the future as more container types are potentially added to the VMI pod.